subsys/shell/shell_dummy.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <shell/shell_dummy.h>
+#include <init.h>
+
+SHELL_DUMMY_DEFINE(shell_transport_dummy);
+SHELL_DEFINE(dummy_shell, "~$ ", &shell_transport_dummy, 10,
+	     SHELL_FLAG_OLF_CRLF);
+
+static int init(const struct shell_transport *transport,
+		const void *config,
+		shell_transport_handler_t evt_handler,
+		void *context)
+{
+	struct shell_dummy *sh_dummy = (struct shell_dummy *)transport->ctx;
+
+	if (sh_dummy->initialized) {
+		return -EINVAL;
+	}
+
+	sh_dummy->initialized = true;
+
+	return 0;
+}
+
+static int uninit(const struct shell_transport *transport)
+{
+	struct shell_dummy *sh_dummy = (struct shell_dummy *)transport->ctx;
+
+	if (sh_dummy->initialized == false) {
+		return -EINVAL;
+	}
+
+	sh_dummy->initialized = false;
+
+	return 0;
+}
+
+static int enable(const struct shell_transport *transport, bool blocking)
+{
+	struct shell_dummy *sh_dummy = (struct shell_dummy *)transport->ctx;
+
+	if (sh_dummy->initialized == false) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int write(const struct shell_transport *transport,
+		 const void *data, size_t length, size_t *cnt)
+{
+	struct shell_dummy *sh_dummy = (struct shell_dummy *)transport->ctx;
+
+	if (!sh_dummy->initialized) {
+		*cnt = 0;
+		return -EINVAL;
```

Comment:
i would return -ENODEV but it's a minor suggestions as there is no code that would be a perfect match.